app/services/user_access_key.rb

```diff
@@ -75,13 +75,7 @@ def made?
   attr_accessor :made, :unlocked
 
   def z1_with_padding(len)
-    cur_len = z1.length
-    str = z1.dup
-    while cur_len < len
-      str = '0' + str
-      cur_len += 1
-    end
-    str
+    z1.dup.rjust(len, '0')
```

Comment:
super tiny comment: possible to use `length` instead of `len`? I read this a couple of times wondering what `len` meant but then I visited the rjust docs and realized LENGTH  https://ruby-doc.org/core-2.2.3/String.html#method-i-rjust